batcher/aligned-sdk/src/communication/batch.rs

```diff
@@ -0,0 +1,136 @@
+use tokio::time::timeout;
+
+use crate::{
+    core::errors,
+    core::types::{
+        AlignedVerificationData, BatchInclusionData, VerificationCommitmentBatch,
+        VerificationDataCommitment,
+    },
+    eth::BatchVerifiedEventStream,
+};
+
+const AWAIT_BATCH_VERIFICATION_TIMEOUT: u64 = 60;
+
+pub async fn handle_batch_inclusion_data<'s>(
+    batch_inclusion_data: BatchInclusionData,
+    aligned_verification_data: &mut Vec<AlignedVerificationData>,
+    verification_data_commitments_rev: &mut Vec<VerificationDataCommitment>,
+    stream: &mut BatchVerifiedEventStream<'s>,
+    verified_batch_merkle_roots: &mut HashSet<Vec<u8>>,
+) -> Result<(), errors::SubmitError> {
+    let _ = handle_batch_inclusion_data_without_await(
+        batch_inclusion_data.clone(),
+        aligned_verification_data,
+        verification_data_commitments_rev,
+    );
+
+    let batch_merkle_root = batch_inclusion_data.batch_merkle_root.to_vec();
+
+    if !verified_batch_merkle_roots.contains(&batch_merkle_root) {
+        await_batch_verification(stream, &batch_inclusion_data.batch_merkle_root).await?;
+        verified_batch_merkle_roots.insert(batch_merkle_root);
+    }
+
+    Ok(())
+}
+
+pub fn handle_batch_inclusion_data_without_await<'s>(
+    batch_inclusion_data: BatchInclusionData,
+    aligned_verification_data: &mut Vec<AlignedVerificationData>,
+    verification_data_commitments_rev: &mut Vec<VerificationDataCommitment>,
+) -> Result<(), errors::SubmitError> {
+    debug!("Received response from batcher");
+    debug!(
+        "Batch merkle root: {}",
+        hex::encode(batch_inclusion_data.batch_merkle_root)
+    );
+    debug!("Index in batch: {}", batch_inclusion_data.index_in_batch);
+
+    let verification_data_commitment =
+        verification_data_commitments_rev.pop().ok_or_else(|| {
+            errors::SubmitError::GenericError("Verification data commitments are empty".to_string())
+        })?;
+
+    if verify_response(&verification_data_commitment, &batch_inclusion_data) {
+        aligned_verification_data.push(AlignedVerificationData::new(
+            &verification_data_commitment,
+            &batch_inclusion_data,
+        ));
+    }
+
+    Ok(())
+}
+
+async fn await_batch_verification<'s>(
+    stream: &mut BatchVerifiedEventStream<'s>,
+    batch_merkle_root: &[u8; 32],
+) -> Result<(), errors::SubmitError> {
+    let await_batch_verified_fut = await_batch_verified_event(stream, batch_merkle_root);
+
+    match timeout(
+        Duration::from_secs(AWAIT_BATCH_VERIFICATION_TIMEOUT),
+        await_batch_verified_fut,
+    )
+    .await
+    {
+        Ok(Ok(_)) => {
+            debug!("Batch operator signatures verified on Ethereum");
+        }
+        Ok(Err(e)) => {
+            error!(
+                "Error awaiting for batch signature verification event: {}",
+                e
+            );
+        }
+        Err(_) => {
+            debug!("Batch operator signatures were not verified yet on Ethereum");
```

Comment:
Similarly, here we should return an error indicating the timeout was reached. The user should know what happened. Maybe if the timeout is reached, they would want to try the `verify` command by themselves or something similar

Reply:
Solved in [cef4afb](https://github.com/yetanotherco/aligned_layer/pull/583/commits/cef4afb1e728da34f45728b42d6780e29a862585)